src/serve/functions.ts

```diff
@@ -18,7 +18,9 @@ module.exports = {
     args = {
       projectId,
       functionsDir,
-      ...args,
+      // TODO: args also contain projectId and functionsDir and thus the fields
+      // above will be overwritten. Why do we even need those?
+      ...(args as object),
```

Comment:
Just want to keep the runtime behavior the same in this PR so it's easier to revert. Let me know if you'd prefer a follow-up PR to fix this (although it's still unclear to me what the "fix" should be).

Reply:
I think the reason for this is because there are some poorly-typed tests which pass around incomplete `FunctionsEmulatorArgs`.

Reply:
Thanks for the context! I've updated the comments to document this.